File: AerisDemo/src/com/example/service/ScreenOnService.java
package com.example.service;

import android.app.Service;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.IBinder;

import com.hamweather.aeris.logging.Logger;

public class ScreenOnService extends Service {

	private static final String TAG = "AERIS.UserPresentService";
	private ScreenOnReceiver receiver;

	/*
	 * (non-Javadoc)
	 * 
	 * @see android.app.Service#onCreate()
	 */
	@Override
	public void onCreate() {
		super.onCreate();
		receiver = new ScreenOnReceiver();
		IntentFilter filter = new IntentFilter(Intent.ACTION_SCREEN_ON);
		filter.setPriority(999);
		registerReceiver(receiver, filter);
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see android.app.Service#onBind(android.content.Intent)
	 */
	@Override
	public IBinder onBind(Intent intent) {
		return null;
	}

	@Override
	public int onStartCommand(Intent intent, int flags, int startId) {
		return Service.START_STICKY;
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see android.app.Service#onDestroy()
	 */
	@Override
	public void onDestroy() {
		Logger.d(TAG, "Screen On Service destroyed...");
		unregisterReceiver(receiver);
		super.onDestroy();
	}
}

File: AerisSdkDemo/app/src/main/java/com/example/demoaerisproject/service/ScreenOnService.kt
package com.example.demoaerisproject.service

import android.app.Service
import android.content.Intent
import android.content.IntentFilter
import android.os.IBinder
import com.aerisweather.aeris.logging.Logger


class ScreenOnService : Service() {
    private var receiver: ScreenOnReceiver? = null

    /*
	 * (non-Javadoc)
	 *
	 * @see android.app.Service#onCreate()
	 */
    override fun onCreate() {
        super.onCreate()
        receiver = ScreenOnReceiver()
        val filter = IntentFilter(Intent.ACTION_SCREEN_ON)
        filter.priority = 999
        registerReceiver(receiver, filter)
    }

    /*
	 * (non-Javadoc)
	 *
	 * @see android.app.Service#onBind(android.content.Intent)
	 */
    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        return START_STICKY
    }

    /*
	 * (non-Javadoc)
	 *
	 * @see android.app.Service#onDestroy()
	 */
    override fun onDestroy() {
        Logger.d(TAG, "Screen On Service destroyed...")
        unregisterReceiver(receiver)
        super.onDestroy()
    }

    companion object {
        private const val TAG = "AERIS.UserPresentService"
    }
}
